src/kixi/heimdall/kaylee.clj

```diff
@@ -53,25 +52,32 @@
     (if (service/add-member-event (db)
                                   (comms)
                                   (:id user) (:id group))
-      (loop []
-        (let [groups (member/retrieve-groups-ids (db) (:id user))]
-          (if (contains? (set groups) (:id group))
-            true
-            (do (Thread/sleep 1000)
-                (recur)))))
+      (wait-for #(let [groups (member/retrieve-groups-ids (db) (:id user))]
+                   (contains? (set groups) (:id group))))
       :failed-add-user-to-group)))
 
+(defn invite-user!
+  ([username name]
+   (invite-user! username name []))
+  ([username name group-names]
```

Comment:
This is a bit confusing tbh. It feels like group creation should be a separate step - not sure there's any benefit to doing it here, all in one fn. Also, you don't need to add users to a group they are the owner of, IIRC.

Reply:
Kinda two points:

1. I added to here to make it super easy for Kaylee to invite users and get them into groups. Could relabel this/add a fn "invite-and-add-to-groups", to make it clearer.
2. Your right you don't, create-group does a cheeky add. Might change it abit so it only does the add if it didn't create the group.